torch/nn/parallel/distributed.py

```diff
@@ -603,3 +654,16 @@ def _passing_sync_batchnorm_handle(self, module_copies):
                     assert self.device_type != 'cpu', "SyncBatchNorm layers only work with GPU modules"
                     layer._specify_ddp_gpu_num(
                         len(self.device_ids) if self.device_ids else 1)
+
+    def _check_comm_hook(self, hook):
```

Comment:
we can add check that the value hold inside Future are tensor vector and size is same as tensors size inside gradBucket

Reply:
I think including this check without running the hook is not possible. Instead I put a warning in ddp comm hook documentation.

Reply:
sounds good